# Throughput

In [1]:
from utils import *
import matplotlib.pyplot as plt

In [2]:
data_paths = ['../results/AR-24-30/AR.dat']
data = None
stat = None
for path in data_paths:
    data, stat = get_data(path)
print(stat)

{'#total-acquire-release:': 5857.0}


In [3]:
avg, std = avg_rate(data, 3)
print('Aerage:', avg)
print('STD:', std)

Aerage: 16339.990492330637
STD: 1272.526032755494


In [4]:
print(5857 * 2)
print(5857 * 2 / 30)
print(1 - 5857 * 2 / 30 / 16163)
print(16163*30 / 5857)

11714
390.46666666666664
0.9758419435335849
82.78811678333618


we know we have 5857 pairs of (acquire-release)  
then we have 5857 * 2 = 11714 writes in total for 30s  
average number of writes per second is 390  
the read percentage is approximately 97.5%  
the implementation is quite inefficient, each aquire-release takes around 83 operations  

next I am going to test with the built-in receipt InterProcessMutex, and compared with our implementation  
I am expecting more overhead in the receipt  

# Latency

In [5]:
prefix = '../results/AR-24-30/'
files = get_latency_files('AR', prefix)
data = {}
for f in files:
    idx = int(f.split('-')[0])
    data[idx], _ = get_data(os.path.join(prefix, f))

print(data.keys())

dict_keys([18, 22, 12, 23, 4, 21, 7, 20, 6, 2, 15, 11, 10, 16, 13, 3, 9, 14, 1, 19, 8, 17, 0, 5])


In [6]:
avg, std = avg_latency(data)
print('Average AR:', avg)
print('STD AR:', std)

Average AR: 0.12312439654157417
STD AR: 0.017987144193201878


In [7]:
prefix = '../results/MUTEX-24-30/'
files = get_latency_files('MUTEX', prefix)
data = {}
for f in files:
    idx = int(f.split('-')[0])
    data[idx], _ = get_data(os.path.join(prefix, f))

print(data.keys())

dict_keys([18, 15, 4, 11, 10, 1, 21, 13, 22, 0, 2, 5, 17, 3, 16, 14, 6, 20, 12, 8, 19, 23, 7, 9])


In [8]:
avg, std = avg_latency(data)
print('Average MUTEX:', avg)
print('STD MUTEX:', std)

Average MUTEX: 0.08123524992900606
STD MUTEX: 0.007262383371640504
